### ISSUE TITLE
systemd-networkd: default ipv6 route created for wireguard network unexpectedly

### ISSUE THREAD

**Issue by stephenmw** [systemd/systemd#27026]
### systemd version the issue has been seen with

systemd-251.4-1ubuntu7

### Used distribution

Ubuntu 22.10

### Linux kernel version used

5.19.0-1015-gcp

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

When I create a network on a Wireguard interface with no configured peers or routes, I expect no routes to be configured.

Wireguard is special in that a link local address is never created by default. My belief is that this is causing some sort of confusion.

### Unexpected behaviour you saw

I am getting a "default" route in ipv6:

```
# ip -6 route | grep wg0
default dev wg0 proto kernel metric 256 pref medium
```
### Steps to reproduce the problem

/etc/systemd/network/wg0.netdev 

```
[NetDev]
Kind=wireguard
Name=wg0

[WireGuard]
ListenPort=51820
PrivateKey=kGxksMmis80n8vX02JESThWQZOUtNP8ED5rr0Z+wMHk=
```

/etc/systemd/network/wg0.network 

```
[Match]
Name=wg0

[Network]
Address=192.168.14.1
Address=fd02:87f6:2f9f::1
```

Reload

```
# networkctl reload
```

After reload, check routes and you will see that ipv6 has a default route.

**Comment by yuwata** [comment]
The wg0 interface is really created with wg0.netdev ? Note, on `networkctl reload`, networkd does not recreate the existing interface.
Please try to remove wg0, then restart networkd.
If that does not work for you, then please enable to generate debugging logs by creating the following drop-in:
```
# /etc/systemd/system/systemd-networkd.service.d/override.conf
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```
then provide the generated logs.

**Comment by stephenmw** [comment]
This was originally seen on NixOS in a real configuration. I created a new GCE instance with Ubuntu to validate the issue and build a minimum reproduction. At no point on the test instance did I create a wireguard link using a tool other than systemd-networkd.

```
root@systemd-test:~# ip link del wg0
root@systemd-test:~# systemctl restart systemd-networkd
root@systemd-test:~# ip -6 route
::1 dev lo proto kernel metric 256 pref medium
fe80::/64 dev ens4 proto kernel metric 256 pref medium
default dev wg0 proto kernel metric 256 pref medium
root@systemd-test:~# journalctl -u systemd-networkd --since "Wed 2023-03-29 14:24:42 UTC" > networkd-restart.log
```

Log: [networkd-restart.log](https://github.com/systemd/systemd/files/11102047/networkd-restart.log)

The most interesting line:

```
Mar 29 14:28:56 systemd-test systemd-networkd[1047]: wg0: Received new foreign route (configured): dst: n/a, src: n/a, gw: n/a, prefsrc: n/a, scope: global, table: main(254), proto: kernel, type: unicast, nexthop: 0, priority: 256, flags: n/a
```

I get the same result running this on NixOS in a real environment with full IPv6 connectivity, peers, and not on GCE. I have already found a workaround which is to set a low metric on my real IPv6 gateway route.

**Comment by yuwata** [comment]
`systemd-networkd` does not add routes with 'kernel' proto. As the debugging log clarified, the route is added by the kernel or other tools. I am not familiar with WireGuard, but I guess a kind of conflicting service is running on the machine.

**Comment by stephenmw** [comment]
I have more information on the issue. This is definitely, at least indirectly, caused by networkd.

First, the issue cannot be reproduced manually:

```
wg setconf wg0 wg0.conf
ip addr add 192.168.14.1 dev wg0
ip addr add fd02:87f6:2f9f::1 dev wg0
ip -6 route
```

wg0.conf (created using `wg showconf wg0 > wg0.conf` from what networkd configured)

```
[Interface]
ListenPort = 51820
PrivateKey = kGxksMmis80n8vX02JESThWQZOUtNP8ED5rr0Z+wMHk=
```

After adding the networkd config files back and reloading, the route reappears. So something it does is causing this.

Second, I found a solution to prevent it. When you add masks to the Addresses in wg0.network, the default route does not appear:

```
[Network]
Address=192.168.14.1/32
Address=fd02:87f6:2f9f::1/128
```

This is a valid solution for me, so you can close out the bug if you wish. However, this seems like something you may want to look into as a papercut.

**Comment by yuwata** [comment]
Hmm... Will take a look.

**Comment by yuwata** [comment]
Duplicate of #26102, already fixed by #26105.
